File: java/src/main/scala/properties/DummyDao.scala
package properties

object DummyDao {
  private val personsDb = collection.mutable.Map.empty[String, Person]

  def insert(k: String, p: Person) = {
    personsDb += k -> p
  }

  def get(k: String): Option[Person] = {
    personsDb.get(k)
  }
}

File: kotlin/src/main/kotlin/properties/DummyDao.kt
package properties

object DummyDao {
    private val mockDb = mutableMapOf<String, Person>()

    fun insert(p: Person) = mockDb.putIfAbsent(p.name, p)

    operator fun get(name: String) = mockDb[name]

    fun size() = mockDb.size
}
